graph/src/blockchain/mod.rs

```diff
@@ -104,7 +106,7 @@ pub trait Blockchain: Debug + Sized + Send + Sync + 'static {
         filter: Self::TriggerFilter,
         metrics: Arc<BlockStreamMetrics>,
         unified_api_version: UnifiedMappingApiVersion,
-    ) -> Result<BlockStream<Self>, Error>;
+    ) -> Result<Box<dyn BlockStream<Self>>, Error>;
```

Comment:
For chain-specific types we've so far chosen to use associated types rather than the trait objects, so to be consistent a `type BlockStream` should be added, and this should be `Self::BlockStream`.

Reply:
`BlockStream` is completely chain agnostic, hence why I did not create an associated type. Both of the implementation are chain agnostic (`PollingBlockStream` and the upcoming `FirehoseBlockStream`).

Personally, I do not think that chain agnostic stuff should be associated type on the chain. Tell me what you think.

Reply:
Ok, that convinces me to go with `dyn`, at least for as long as it covers our needs for this trait.